src/test/java/org/commcare/xml/QueryDataParserTest.java

```diff
@@ -44,6 +46,18 @@ public void testParseValueData_withExclude() throws InvalidStructureException, X
         assertEquals(Collections.emptyList(), queryData.getValues(evalContext));
     }
 
+    @Test
+    public void testParseValueData_withRequired() throws InvalidStructureException, XmlPullParserException,
+            IOException, UnfullfilledRequirementsException {
+        String query = "<data key=\"device_id\" ref=\"instance('session')/session/case_id\""
+                + "required=\"true()\"/>";
+        QueryPromptParser parser = ParserTestUtils.buildParser(query, QueryPromptParser.class);
+        QueryPrompt queryData = parser.parse();
+
+        EvaluationContext evalContext = new EvaluationContext(null, TestInstances.getInstances());
+        assert (boolean) queryData.getRequired().eval(evalContext);
```

Comment:
nit: you can use `assertTrue` instead ([example](https://github.com/dimagi/commcare-core/blob/master/src/test/java/org/commcare/backend/session/test/MarkRewindSessionTests.java#L52))

Reply:
Thanks - I had misunderstood how to import `assertTrue`

c540a4565